protostar/cheatable_starknet/cheatcodes/expect_events_cairo_cheatcode.py

```diff
@@ -0,0 +1,54 @@
+from typing import Callable
+
+from typing_extensions import NotRequired, TypedDict
+
+from protostar.cheatable_starknet.cheatcodes.cairo_cheatcode import CairoCheatcode
+from protostar.cheatable_starknet.cheaters import CairoCheaters
+from protostar.cheatable_starknet.cheaters.expect_events_controller import (
+    ExpectEventsController,
+    Event,
+)
+from protostar.starknet import CairoData, Address
+from protostar.starknet.selector import Selector
+
+
+class RawExpectedEvent(TypedDict):
+    from_address: int
+    name: str
+    data: NotRequired[CairoData]
+
+
+class ExpectEventsCheatcode(CairoCheatcode):
+    def __init__(self, controller: ExpectEventsController, cheaters: "CairoCheaters"):
+        super().__init__(cheaters)
+        self._controller = controller
```

Comment:
I don't think we should pass this one as argument atm, there is no reason to do so. Can we just create it inside cheatcode?

Reply:
No. `ExpectEventsCheatcode` would have inherited `ExpectEventsController` dependencies.

Reply:
You are right, but I am concerned that we have inconsistency between cheatcodes (which lead to my confusion), other cheatcodes are using cheaters to pass those dependencies. Can we create a task for removing Cheaters and consolidating it to your approach?

Reply:
Do we need to create tasks for refactoring? It's hard to forget about it.

Reply:
It doesn't hurt to have it

Reply:
We definiately need to for parts which we decided to refactor. We are using tasks as async communication mechanism. Keeping stuff as tribe knowledge is always a bad idea

Reply:
https://github.com/software-mansion/protostar/issues/1440